### ISSUE TITLE
remove outbound-rtp totalEncodedBytesTarget

### ISSUE THREAD

**Issue by fippo** [w3c/webrtc-stats#653]
https://w3c.github.io/webrtc-stats/#dom-rtcoutboundrtpstreamstats-totalencodedbytestarget
is implemented by libwebrtc but returns 0. So far nobody has complained (but me but I complain about *everything*)

This doesn't offer much value over comparing targetBitrate to 8*bytesSent/second
The difference is RTP packetization overhead which is a few bytes per RTP packet (or frame) which is depending on the codec and not modifiable in any way.

Additionally it poses some extra problems for encoded-transform. Since that operates on frames, adding encryption bytes to it should be accounted for in the totalEncodedBytesTarget. Which is a major headache given how this is currently implemented in libwebrtc at least.

Please vote with 👎 to remove or 👍 to keep

**Comment by henbos** [comment]
This was implemented at one point, so before reming it I would try to look into why it stopped working. Perhaps it broke in a refactoring and is easy to fix.

**Comment by henbos** [comment]
I think it got implemented in 2019 [here](https://webrtc-review.googlesource.com/c/src/+/137200) and that it worked, but when implementing simulcast in 2020 [here](https://webrtc-review.googlesource.com/c/src/+/174480) and total_encoded_bytes_target was supposedly moved from "per sender" VideoSendStream::Stats to "per layer" VideoSendStream::StreamStats it wasn't actually moved, just added and never touched. So we continued to set the value on the "per sender" stat but we began reading from the never-set "per layer" stat. The dangers of not using absl::optional<>.

I suggest filing a crbug since this may be easy to fix.

**Comment by fippo** [comment]
true, being 0 is indeed a regressio between M84 where it [broke](https://chromium.googlesource.com/chromium/src/+/0a86fece67caddb67d91a096d97688c0b2087b97) and current M104.

**Comment by dontcallmedom** [comment]
@henbos / @fippo can one of you file the crbug and close this issue then?

**Comment by fippo** [comment]
the question whether it adds value over targetBitrate (implemented in Chrome as of today) remains

**Comment by henbos** [comment]
See https://github.com/w3c/webrtc-stats/issues/416 for motivation for totalEncodedBytesTarget despite targetBitrate existing at the time of adding it

**Comment by fippo** [comment]
@henbos: seems you are now supportive https://issues.webrtc.org/issues/42223582#comment10

**Comment by henbos** [comment]
I am in favor of deleting it because it has been broken for so long and is very similar to targetBitrate, plus based on recent conversations @youennf is wary of us having too many metrics in getStats, so I'm sure he'd be supportive too

**Comment by henbos** [comment]
Backwards compat-wise it would seem hard to imagine someone depending on this existing :P
